File: labs/calculator/calculator.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// adds/subtracts/multiplies all values that are in the *values array.
// nValues is the number of values you're reading from the array
// operator will indicate if it's an addition (1), subtraction (2) or
// multiplication (3)
long calc(int operator, int nValues, int *values) {
    int result = values[0];
    int i;
    if(operator == 1){
        printf("%d",result);
        for(i = 1; i<nValues; i++){
            printf(" + %d",values[i]);
            result += values[i];
        }
        return result;

    }else if (operator == 2){
        printf("%d",result);
        int tmp = 0;
        for(i = 1; i<nValues; i++){
            printf(" - %d",values[i]);
            tmp += values[i];
        }
        result = tmp - result;
        return result;

    }else if (operator == 3){
        printf("%d",result);
        for(i = 1; i<nValues; i++){
            printf(" * %d",values[i]);
            result*= values[i];
        }
        return result;

    }else{
        printf("Something went wrong, try again\n");
    }
    
    return 0;
}



int main(int argc, char **argv) {
    char str1[] = "add", str2[] = "sub", str3[] = "mult";
    char *ptr;
    int *arr;
    arr = malloc(sizeof(int)*argc-1);
    int j;
    long res;
    int oper;
    int f = 0;
    if(strcmp(str1, argv[1])==0){
        oper = 1;
    }else if(strcmp(str2, argv[1])==0){
        oper = 2;
    }else if(strcmp(str3, argv[1])==0){
        oper = 3;
    }else{
            printf("Need to put add, sub or mult\n");
            return -1;
    }
    
    for(j=0;j<argc-2;j++){
        if(strtol(argv[2+j], &ptr, 10)!=0){
            arr[j]=atoi(argv[2+j]);
            f++;
        }else{
           printf("No strings, only numbers\n"); 
           return -1;
        }
    }
    res = calc(oper, f, arr);
    if(oper == 2){
        printf(" = -%lu\n",res);
    }
    else{
        printf(" = %lu\n",res);
    }
    return -1;
}

File: labs/go-calculator/calculator.go
package main

// adds/subtracts/multiplies all values that are in the *values array.
// nValues is the number of values you're reading from the array(?)
// operator will indicate if it's an addition (1), subtraction (2) or
// multiplication (3)

import (
	"fmt"
	"os"
	"strconv"
)

func calc(operator int, values []int) int {
	r := values[0]
	if operator==1{
		fmt.Printf("%d",r)
		for i := 1;i<len(values);i++{
			fmt.Printf(" + %d",values[i])
			r += values[i]
		}
	}else if operator == 2{
		fmt.Printf("%d",r)
		for i := 1;i<len(values);i++{
			fmt.Printf(" - %d",values[i])
			r -= values[i]
		}
	}else if operator == 3{
		fmt.Printf("%d",r)
		for i := 1;i<len(values);i++{
			fmt.Printf(" * %d",values[i])
			r *= values[i]
		}
	}
	return r
}

func main() {
	var arr [256]int
	var oper int
	op := os.Args[1]
	if op == "add"{
		oper = 1
	}else if op == "sub"{
		oper = 2
	} else if op == "mult"{
		oper = 3
	}else{
		fmt.Println("Need to put add, sum or mult as the operator")
		os.Exit(1)
	}
	for i:= 2; i<len(os.Args) ;i++{
		f,err := strconv.Atoi(os.Args[i])
		if err != nil{
			fmt.Println("Only integers accepted")
			os.Exit(3)
		}
		arr[i-2] = f
	}
	slice := arr[0:len(os.Args)-2]
	result := calc(oper, slice)
	fmt.Printf(" = %d\n", result)
}